buzzmobile/tests/test_utils/rostest_utils.py

```diff
@@ -43,14 +43,13 @@ def await(gen):
 
         await(TestNode.wait_for_message())
 
-    it will wait for the `wait_for_message` call to yield the first (and only)
-    time. This will happen only if the node has received a message
-    (`self.recieved == True`), which means that you can then safely poll for
-    the recieved message (`TestNode.message.data` or otherwise).
+    it will wait for the `wait_for_message` call to finish running.
 
     In other words, if await(Node) doesn't raise an error, you should have
     successfully recieved a message, and can therefore test against it,
     otherwise the message access can raise a NoMessage error.
     """
+    res = None
     for item in gen:
-        yield item
+        res = item
```

Comment:
I think it could just as well be

```
for _ in gen:
    _
```
but I wanted to be able to return the result. The yield change is important. I was being dumb before.